src/meta/api/src/share_api_impl.rs

```diff
@@ -1289,3 +1355,68 @@ async fn drop_all_database_from_share(
     }
     Ok(())
 }
+
+async fn convert_share_meta_to_spec(
+    kv_api: &(impl KVApi + ?Sized),
+    share_name: &str,
+    share_meta: ShareMeta,
+) -> MetaResult<ShareSpec> {
+    let database = if let Some(database) = share_meta.database {
+        if let ShareGrantObject::Database(db_id) = database.object {
+            let id_key = DatabaseIdToName { db_id };
+
+            let (_db_meta_seq, db_name): (_, Option<DatabaseNameIdent>) =
+                get_struct_value(kv_api, &id_key).await?;
+            if let Some(db_name) = db_name {
+                Some(ShareDatabaseSpec {
+                    name: db_name.db_name,
+                    id: db_id,
+                })
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    } else {
+        None
+    };
+
+    let mut tables = vec![];
+    for (_, entry) in share_meta.entries.iter() {
+        if let ShareGrantObject::Table(table_id) = entry.object {
+            let tbid = TableId { table_id };
+
+            let (_tb_meta_seq, tb_meta): (_, Option<TableMeta>) =
+                get_struct_value(kv_api, &tbid).await?;
+            if let Some(tb_meta) = tb_meta {
+                let options = &tb_meta.options;
+                let location = options
```

Comment:
is this the location the last snapshot of table or the prefix of table?

Reply:
we'd prefer a path in a directory manner, thus we can check whether a path is inside the prefix,  no matter what type of this file